packages/html-reporter/src/filter.ts

```diff
@@ -118,12 +125,16 @@ export class Filter {
       if (!matches)
         return false;
     }
-
     if (this.text.length) {
       const matches = this.text.filter(t => searchValues.text.includes(t)).length === this.text.length;
       if (!matches)
         return false;
     }
+    if (this.labels.length) {
+      const matches = this.labels.filter(l => searchValues.text.match(new RegExp(`(\\s|^)${l}(\\s|$)`, 'g'))).length === this.labels.length;
```

Comment:
`searchValues.text` has just been constructed to contain status, project name, path, title. You don't really want to regex against it. What you probably want it to parse out labels from the test.path + test.title into `searchValues.labels` and check that all the filter labels are present in the `searchValues.labels`.